web/src/components/navbar/NavBarPopupMenu.test.tsx

```diff
@@ -0,0 +1,281 @@
+import { ROUTES } from "@/lib/domain-logic/routes";
+import { withAuth } from "@/test/helpers/helpers-renderers";
+import { mockPush, useMockRouter } from "@/test/mock/mockRouter";
+import { useMockUserData } from "@/test/mock/mockUseUserData";
+import { setupStatefulUserDataContext, WithStatefulUserData } from "@/test/mock/withStatefulUserData";
+import {
+  BusinessUser,
+  generateProfileData,
+  generateUser,
+  generateUserData,
+  UserData,
+} from "@businessnjgovnavigator/shared";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+
+const Config = getMergedConfig();
+
+const mockApi = api as jest.Mocked<typeof api>;
+
+jest.mock("next/router", () => ({ useRouter: jest.fn() }));
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+jest.mock("@/lib/api-client/apiClient", () => ({ postSelfReg: jest.fn() }));
+jest.mock("@/lib/auth/signinHelper", () => {
+  const originalSigninHelper = jest.requireActual("@/lib/auth/signinHelper");
+  return {
+    ...originalSigninHelper,
+    onSignOut: jest.fn(),
+  };
+});
+jest.mock("@/lib/auth/sessionHelper", () => ({
+  triggerSignIn: jest.fn(),
+}));
+
+const generateOnboardingUser = (): UserData => {
+  return generateUserData({
+    profileData: generateProfileData({
+      businessName: "",
+      industryId: undefined,
+      legalStructureId: undefined,
+    }),
+  });
+};
+
+const businessName = "businessName";
+
+const generateBusinessNamedUserData = (overrides?: Partial<UserData>): UserData => {
+  return generateUserData({
+    profileData: generateProfileData({
+      businessName: businessName,
+      industryId: "cannabis",
+      legalStructureId: "limited-liability-company",
+    }),
+    ...overrides,
+  });
+};
+
+describe("<NavBarPopupMenu />", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    useMockRouter({});
+  });
+
+  const renderNavBarPopupMenu = (
+    overrides: Partial<NavBarPopupMenuProps>,
+    userData?: UserData,
+    user?: BusinessUser
+  ): void => {
+    render(
+      withAuth(
+        <WithStatefulUserData initialUserData={userData}>
+          <NavBarPopupMenu
+            handleClose={(): void => {}}
+            open={true}
+            hasCloseButton={false}
+            menuConfiguration={"login"}
+            {...overrides}
+          />
+        </WithStatefulUserData>,
+        { user, isAuthenticated: IsAuthenticated.FALSE }
+      )
+    );
+  };
+
+  it("renders close button when hasCloseButton prop is true", () => {
+    useMockUserData({});
+    renderNavBarPopupMenu({ hasCloseButton: true });
+    expect(screen.getByTestId("close-button-nav-menu")).toBeInTheDocument();
+  });
+
+  it("triggers handleClose when close button is pressed", () => {
+    useMockUserData({});
+    renderNavBarPopupMenu({ hasCloseButton: false });
+    expect(screen.queryByTestId("close-button-nav-menu")).not.toBeInTheDocument();
```

Comment:
I don't think this test name matches what is being tested

FYI, if you *did* want to test that it triggers handleClose (seems like a good idea, in addition to this close button false prop test here), the way to do that would be to pass a jest mock function into handleClose prop on line 78 and then expect it to have been called

Reply:
Yep yep yep, I totally wrote then and then meant to but forgot to come back to it